### PR TITLE
refactor :  modular auth

### PR DESCRIPTION
**Pull request by mimansha-swarup**
- auth as taught in yesterday's class

### CODE REVIEW

**Review by preethasri — APPROVED** [review]
great work. good to merge👍

**Review by ayush65 — APPROVED** [review]
code looks clean 
 good file structure 
ClassNames are self Explanatory

File: src/Routes/AllRoutes.js [hunk 1]
```diff
@@ -1,30 +1,33 @@
...
-      <Route path="/mockman" element={<Mockman />} />
-      <Route path="/playlist" element={<PlayListPage />} />
-      <Route path="/playlist/:playlistId" element={<SinglePlaylistPage />} />
+      <Route path="/video" element={<RequiresAuth><VideoPage /></RequiresAuth>} />
+      <Route path="/liked-video" element={<RequiresAuth><LikedPage /></RequiresAuth>} />
+      <Route path="/video/:videoId" element={<RequiresAuth> <SingleVideoPage /> </RequiresAuth>} />
+      <Route path="/watch-later" element={<RequiresAuth> <WatchLaterPage /> </RequiresAuth>} />
+      <Route path="/history" element={<RequiresAuth> <HistoryPage /> </RequiresAuth>} />
+      <Route path="/playlist" element={<RequiresAuth> <PlayListPage /> </RequiresAuth>} />
+      <Route path="/playlist/:playlistId" element={<RequiresAuth> <SinglePlaylistPage /> </RequiresAuth>} />
```
Comment (kshitijthatte) [inline src/Routes/AllRoutes.js:23]: You can wrap all the protected routes in a single `RequiresAuth`, checkout `<Outlet />` in `react-router`

**Review by kshitijthatte — APPROVED** [review]

File: src/Routes/AllRoutes.js (hunk 1 above)
Comment (mimansha-swarup) [reply src/Routes/AllRoutes.js:23]: ok will check late

**Pull request merged**